File: leetcode/tree/sum_root_to_leaf_numbers.go
package main

import "fmt"

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func sumNumbers(root *TreeNode) int {
	if root == nil {
		return 0
	}
	var nums [][]int
	var arr []int
	helper(root, arr, &nums)

	sum := 0
	for i := 0; i < len(nums); i++ {
		sum += sliceToInt(nums[i])
	}
	return sum
}

func sliceToInt(s []int) int {
	res := 0
	op := 1
	for i := len(s) - 1; i >= 0; i-- {
		res += s[i] * op
		op *= 10
	}
	return res
}

func helper(root *TreeNode, arr []int, out *[][]int) {
	tmp := make([]int, len(arr))
	copy(tmp, arr)
	tmp = append(tmp, root.Val)
	if root.Left == nil && root.Right == nil {
		*out = append(*out, tmp)
		return
	}
	if root.Left != nil {
		helper(root.Left, tmp, out)
	}
	if root.Right != nil {
		helper(root.Right, tmp, out)
	}
}

func main() {
	root := &TreeNode{4, nil, nil}
	root.Left = &TreeNode{9, nil, nil}
	root.Right = &TreeNode{0, nil, nil}
	root.Left.Left = &TreeNode{5, nil, nil}
	root.Left.Right = &TreeNode{1, nil, nil}
	fmt.Println(sumNumbers(root))
}

File: leetcode/tree/sum_root_to_leaf_numbers.py
from common import TreeNode
from copy import deepcopy

class Solution:
    def sumNumbers(self, root: TreeNode) -> int:
        if not root:
            return 0
        nums = []
        def helper(root, num):
            tmp = deepcopy(num)
            tmp.append(root.val)
            if not root.left and not root.right:
                nums.append(tmp)
                return

            if root.left:
                helper(root.left, tmp)
            if root.right:
                helper(root.right, tmp)
        helper(root, [])
        sum = 0
        for num in nums:
            s = [str(i) for i in num]
            sum += int("".join(s)) 
        return sum            

if __name__ == "__main__":
    root = TreeNode(4)
    root.left = TreeNode(9)
    root.right = TreeNode(0)
    root.left.left = TreeNode(5)
    root.left.right = TreeNode(1)
    print(Solution().sumNumbers(root))